daemon/daemon.go

```diff
@@ -877,19 +877,20 @@ func (daemon *Daemon) ShutdownTimeout() int {
 	// By default we use daemon's ShutdownTimeout.
```

Comment:
I notice the GoDoc for the function is not really matching reality: "and is limited by daemon's ShutdownTimeout" doesn't seem right, because it's limited by whatever is longest.

Perhaps something like:

```go
// ShutdownTimeout returns the timeout (in seconds) before containers are forcibly
// killed during shutdown. The default timeout can be configured both on the daemon
// and per container, and the longest timeout will be used. A grace-period of
// 5 seconds is added to the configured timeout.
//
// A negative (-1) timeout means "indefinitely", which means that containers
// are not forcibly killed, and the daemon shuts down after all containers exit.
```

(may need some tweaking, and we should check the documentation as well)

Reply:
ping @allencloud ^^